### PR TITLE
Jetpack Cloud: Update the design of Jetpack Scan section based on the latest styles

### PR DESCRIPTION
**Pull request by rcanepa**
#### Changes proposed in this Pull Request

- Make the Jetpack Scan section match designs from our Figma file.

See 1151678672052943-as-1170456139910593.

##### Okay state

- Navigate to http://jetpack.cloud.localhost:3000/scan/<siteSlug>?scan-state=okay
- Make sure styles match our Figma file

##### Scanning state

- Navigate to http://jetpack.cloud.localhost:3000/scan/<siteSlug>?scan-state=scanning
- Make sure styles match our Figma file

##### Threats state

- Navigate to http://jetpack.cloud.localhost:3000/scan/<siteSlug>?scan-state=threats
- Make sure styles match our Figma file

##### Error state

- Navigate to http://jetpack.cloud.localhost:3000/scan/<siteSlug>?scan-state=error
- Make sure styles match our Figma file

#### Screenshots
image
image
image
image
image

### CODE REVIEW

File: client/landing/jetpack-cloud/components/scan-history-item/style.scss [hunk 1]
```diff
@@ -41,7 +47,7 @@
 		}
 	}
 
-	&__badge {
+	&__badge.badge {
```
Comment (enejb) [inline client/landing/jetpack-cloud/components/scan-history-item/style.scss:50]: I noticed that this badge is not quite centered in height. 
Updating the line-height from 17 -> 14px fixed that for me. 

Before:
Screen Shot 2020-04-10 at 10 16 55 AM

After:
Screen Shot 2020-04-10 at 10 16 35 AM

```suggestion
	&__badge.badge {
	line-height: 14px;
```

**Comment by enejb** [comment]
Nice work here. 

I noticed a couple of thing. 
1. The Fix all button turns gray on hover.
See:
Screen Shot 2020-04-10 at 10 20 42 AM

And the minor thing with the badge not being vertically centered. 
Otherwise I think this looks really good. 

I think it would be good to make some of the changes that are here in this PR more global.
But I think that could be done in future PRs. :)

**Comment by rcanepa** [comment]
I pushed two commits that fix:

- Badge vertical alignment.
- Fix all button styles.

Demo:
image

Kapture 2020-04-10 at 12 51 34

**Comment by keoshi** [comment]
The buttons look much better now!

Though I guess this had some unintentional side-effects:

image

image

image

We probably need to restrict these styles to `.main` or `.layout__primary`, and add styles/classes for “invisible” buttons (which wouldn't have a background or borders).

Finally, I agree with you on the location — maybe we should create a JPC-global `style.scss` in `client/landing/jetpack-cloud` and use it for common styles?

Thanks for making those changes, appreciate it!

**Comment by rcanepa** [comment]
Hey @keoshi !

> We probably need to restrict these styles to .main or .layout__primary, and add styles/classes for “invisible” buttons (which wouldn't have a background or borders).

Sure. The idea is to avoid side-effects on the side bar and other components?

> Finally, I agree with you on the location — maybe we should create a JPC-global style.scss in client/landing/jetpack-cloud and use it for common styles?

Great suggestion. I'll move the styles to that location.

Two questions:

a) The Jetpack Cloud Style Guide doesn't include a button like this one:
image

It resembles the secondary but it has a green border. Is this an exception or could it appear on other places as well? I wonder if we should add styles for this button in the new stylesheet we are creating.

b) What is an invisible button? Do you have an example?

**Comment by keoshi** [comment]
@rcanepa Thanks for noticing that, we hadn't so kudos for bringing it up. A bit late for sure, but we've ran an impromptu poll on p1586793350312400-slack-jetpack-design and the majority agrees that we should follow the pattern existing on the Scan mocks.

I've updated the [Figma Style Guide file](https://www.figma.com/file/r9c7MPvvxT8QLLP8qr25Hc/Jetpack-Cloud-Style-Guide?node-id=923%3A1891&viewport=-3803%2C388%2C1) and will be jumping into this PR tomorrow morning to push those changes.

Thanks again!

**Comment by keoshi** [comment]
> The idea is to avoid side-effects on the side bar and other components?

Yes. Do you think it's a bit severe to nest them under `.main` for example? Since all of our content/visuals live in that section, it would be an easier way to style everything knowing it won't bleed over.

**Comment by keoshi** [comment]
Quick update @rcanepa: pushed a couple of changes to this PR, let me know if they look good to you.

**Comment by rcanepa** [comment]
I looks much nicer now! Thanks for adding those changes.

> I've updated the Figma Style Guide file and will be jumping into this PR tomorrow morning to push those changes.

Got it!

> Yes. Do you think it's a bit severe to nest them under .main for example? Since all of our content/visuals live in that section, it would be an easier way to style everything knowing it won't bleed over.

Yup. I'll change that. However, I noticed one problem with this approach. Elements inside modals are outside `.main`, which means that we can't target them with `.theme-jetpack-cloud .main` (the same happens with `.layout__primary`). This is how the server credential form buttons look like:

image

To fix that, I added the `.theme-jetpack-cloud .ReactModalPortal` selector. Let me know what you think!

**Comment by enejb** [comment]
In my testing this worked really well. 
Nicely done @rcanepa and @keoshi

**Comment by keoshi** [comment]
> To fix that, I added the `.theme-jetpack-cloud .ReactModalPortal` selector. Let me know what you think!

Nice, @rcanepa! Sounds like a solid approach, we just have to keep an eye for elements that are outside of these two, but I imagine there shouldn't be any in the near future.

**Pull request merged**